src/lib/rest/rest.cpp

```diff
@@ -616,6 +630,17 @@ static int contentTypeCheck(ConnectionInfo* ciP)
     return 1;
   }
 
+
+  // Case 5
+  if ((ciP->apiVersion == "v2") && (ciP->httpHeaders.contentType != "application/json"))
+  {
```

Comment:
I'm a fan of using `#define` for all the constant tokens in the token :) However, in this particular case, using APPLICATION_JSON instead of "application/json" could be a bit confusing... I don't have a strong opinion on any side.